### DISCUSSION TITLE
Solution interpolation

### DISCUSSION THREAD

**Discussion by radlicka** [gridap/Gridap.jl#667] [Q&A] [answered]
Can you please provide a simple example for interpolation of FEM solution (e.q. in the case of example 1) . All of the examples except example 13 end with the export of the solution to vtk file. But sometimes you need to work with the solution in Julia script. It would be very helpful, as getting through the code is a quite difficult task ...

Tomas

**Comment by fverdugo** [comment]
Hi  @radlicka !

`v = get_free_dof_values(uh)` gives a plain Julia vector with the dof values of your solution `uh`. Is this what you need?

**Reply by radlicka** [reply]
No, this is not what I need. I need to know the value of the solution (or gradient) in a given point (x,y,z), which need not be a nodal point of the mesh. More specifically, I solve the axially symmetrical Laplace equation (2D problem:  ∇(r ⋅∇ u ) = 0) and I need to know the potential on a line parallel to the axis of symmetry (r=0).  Recently I have everything except the final solution interpolation.

**Comment by ericneiva** [comment]
Hi @radlicka,

You can take a look at this [interpolation tutorial](https://github.com/Balaje/Tutorials/blob/interpolation_tutorial/src/interpolation_fe.jl) - soon available in Gridap tutorials -, which deals with how to interpolate FE functions in arbitrary points (Need at least Gridap v.0.16.4).

**Comment by fverdugo** [comment, marked as answer]
@radlicka you can evaluate the solution `uh` at a given arbitrary `x::Point` object, `uh(x)`.

**Important performance warning**

If you want to evaluate `uh` at several points `x1`,... , `xN` , you have 2 options to perform this operation efficiently:

- **[recommended]** Store these points in a vector `points= [ x1,... ,xN]`, and then evaluate `uh(points)`
-  Build a cache object to reuse data between evaluations:
```julia
cache = return_cache(uh,x1)
evaluate!(cache,uh,x1)
evaluate!(cache,uh,x2)
...
evaluate!(cache,uh,xN)

```

**Reply by radlicka** [reply]
Perfect, thanks for a prompt answer. Tomas
